Documentation/README.md

```diff
@@ -245,13 +243,33 @@ The visibility is applied in following order:
 3. Apply the `Exceptions`
 
 #### Coloring
-The `Coloring` element lists colors and a list of associated components that should be displayed with the specified color when displaying a viewpoint. The color is given in ARGB format. Colors are represented as 6 or 8 hexadecimal digits. If 8 digits are present, the first two represent the alpha (transparency) channel. For example, `40E0D0` would be the color <span style="color:#40E0D0;";>Turquoise</span>. [More information about the color format can be found on Wikipedia.](https://en.wikipedia.org/wiki/RGBA_color_space)
+The `Coloring` element allows specifying the color of components. For each color a list of components to be displayed with the that color should be provided. 
+
+The color is given in ARGB format. Colors are represented as 6 or 8 hexadecimal digits. If 8 digits are present, the first two represent the alpha (transparency) channel. For example, `40E0D0` would be the color Turquoise. [More information about the color format can be found on Wikipedia.](https://en.wikipedia.org/wiki/RGBA_color_space)
+
 
 **Optimization Rules**
 
 BCF is suitable for coloring a few components. A huge list of components causes poor performance. All clients should follow this rule:
 
-* If the size of colored components is huge (approximately 1000 components), alert the user and give him the opportunity to modify the coloring.
+* If the size of colored components is huge (over 1000 components), alert the user and ask them to reduce the number of colored components.
+
+#### Encoding composite Components in selection, visibility and coloring
+
+In IFC, it is [specified](https://standards.buildingsmart.org/documents/Implementation/IFC_Implementation_Agreements/CV-2x3-119.html) that "Geometry for decomposed elements shall be either at the element container or at the element part level". 
+This allows the encoding of viewpoints to be optimized by the guidelines below. The guidelines apply wherever components are listed e.g. `exceptions` in visibility and `components` in selection and coloring.
+
+When selecting, hiding, showing or coloring decomposed components, such as curtain wall or assemblies the following rules (using coloring as an example) should be applied: 
+
+##### When creating the viewpoint
+
+* If the container element and all its parts have the same color then only the container element should be added to the component list
+* If only some parts have the same color, then only those parts should be added to the component list.
+* Adding all the parts to the component list is allowed and would result in excluding any parts that will be added in later revisions of the model
+
+##### When visualizing the viewpoint
+
+The color of a container element should be applied to all its parts.
```

Comment:
This text feels a bit made for "coloring" only.

Reply:
It says on line 262 "using coloring as an example". I think it's easier to understand when it's specific, do you agree?